PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:8 + add:0, change matches:
-    _inherit = ['calendar.event']
-    _inherit = ['hr.department']
-    _inherit = ['hr.employee']
-    _inherit = ['mail.message.subtype']
-    _inherit = ['res.partner']
-    _inherit = ["res.users"]
-    _inherit = ["resource.calendar.leaves"]
-    _inherit = ["resource.calendar"]

Total Changes: 23

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_holidays/models/calendar_event.py
+++ b/addons/hr_holidays/models/calendar_event.py
@@ -2,7 +2,7 @@ from odoo import models
 
 
 class CalendarEvent(models.Model):
-    _inherit = ['calendar.event']
+    _inherit = 'calendar.event'
 
     def _need_video_call(self):
         """ Determine if the event needs a video call or not depending

--- a/addons/hr_holidays/models/hr_department.py
+++ b/addons/hr_holidays/models/hr_department.py
@@ -9,8 +9,7 @@ import ast
 
 
 class HrDepartment(models.Model):
-
-    _inherit = ['hr.department']
+    _inherit = 'hr.department'
 
     absence_of_today = fields.Integer(
         compute='_compute_leave_count', string='Absence by Today')

--- a/addons/hr_holidays/models/hr_employee.py
+++ b/addons/hr_holidays/models/hr_employee.py
@@ -9,7 +9,7 @@ from odoo import _, api, fields, models
 
 
 class HrEmployee(models.Model):
-    _inherit = ['hr.employee']
+    _inherit = 'hr.employee'
 
     current_leave_id = fields.Many2one('hr.leave.type', compute='_compute_current_leave', string="Current Time Off Type",
                                        groups="hr.group_hr_user")

--- a/addons/hr_holidays/models/hr_employee_base.py
+++ b/addons/hr_holidays/models/hr_employee_base.py
@@ -11,7 +11,7 @@ from odoo.addons.resource.models.utils import HOURS_PER_DAY
 
 
 class HrEmployeeBase(models.AbstractModel):
-    _inherit = ["hr.employee.base"]
+    _inherit = "hr.employee.base"
 
     leave_manager_id = fields.Many2one(
         'res.users', string='Time Off',

--- a/addons/hr_holidays/models/hr_leave_accrual_plan.py
+++ b/addons/hr_holidays/models/hr_leave_accrual_plan.py
@@ -11,6 +11,7 @@ DAY_SELECT_SELECTION_NO_LAST = tuple(zip(DAY_SELECT_VALUES, (str(i) for i in ran
 
 
 class HrLeaveAccrualPlan(models.Model):
+    _name = 'hr.leave.accrual.plan'
     _description = "Accrual Plan"
 
     active = fields.Boolean(default=True)

--- a/addons/hr_holidays/models/hr_leave_accrual_plan_level.py
+++ b/addons/hr_holidays/models/hr_leave_accrual_plan_level.py
@@ -17,6 +17,7 @@ def _get_selection_days(self):
 
 
 class HrLeaveAccrualLevel(models.Model):
+    _name = 'hr.leave.accrual.level'
     _description = "Accrual Plan Level"
     _order = 'sequence asc'
 

--- a/addons/hr_holidays/models/hr_leave_mandatory_day.py
+++ b/addons/hr_holidays/models/hr_leave_mandatory_day.py
@@ -5,6 +5,7 @@ from odoo import fields, models
 
 
 class HrLeaveMandatoryDay(models.Model):
+    _name = 'hr.leave.mandatory.day'
     _description = 'Mandatory Day'
     _order = 'start_date desc, end_date desc'
 

--- a/addons/hr_holidays/models/hr_leave_type.py
+++ b/addons/hr_holidays/models/hr_leave_type.py
@@ -19,6 +19,7 @@ _logger = logging.getLogger(__name__)
 
 
 class HrLeaveType(models.Model):
+    _name = 'hr.leave.type'
     _description = "Time Off Type"
     _order = 'sequence'
 

--- a/addons/hr_holidays/models/mail_message_subtype.py
+++ b/addons/hr_holidays/models/mail_message_subtype.py
@@ -9,7 +9,7 @@ _logger = logging.getLogger(__name__)
 
 
 class MailMessageSubtype(models.Model):
-    _inherit = ['mail.message.subtype']
+    _inherit = 'mail.message.subtype'
 
     def _get_department_subtype(self):
         return self.search([

--- a/addons/hr_holidays/models/res_partner.py
+++ b/addons/hr_holidays/models/res_partner.py
@@ -5,7 +5,7 @@ from odoo.addons.mail.tools.discuss import Store
 
 
 class ResPartner(models.Model):
-    _inherit = ['res.partner']
+    _inherit = 'res.partner'
 
     def _compute_im_status(self):
         super(ResPartner, self)._compute_im_status()

--- a/addons/hr_holidays/models/res_users.py
+++ b/addons/hr_holidays/models/res_users.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models, Command
 
 
 class ResUsers(models.Model):
-    _inherit = ["res.users"]
+    _inherit = "res.users"
 
     leave_manager_id = fields.Many2one(related='employee_id.leave_manager_id')
     show_leaves = fields.Boolean(related='employee_id.show_leaves')

--- a/addons/hr_holidays/models/resource.py
+++ b/addons/hr_holidays/models/resource.py
@@ -8,7 +8,7 @@ from datetime import datetime
 
 
 class ResourceCalendarLeaves(models.Model):
-    _inherit = ["resource.calendar.leaves"]
+    _inherit = "resource.calendar.leaves"
 
     holiday_id = fields.Many2one("hr.leave", string='Time Off Request')
 
@@ -156,7 +156,7 @@ class ResourceCalendarLeaves(models.Model):
 
 
 class ResourceCalendar(models.Model):
-    _inherit = ["resource.calendar"]
+    _inherit = "resource.calendar"
 
     associated_leaves_count = fields.Integer("Time Off Count", compute='_compute_associated_leaves_count')
 
